src/ProjectOrigin.WalletSystem.Server/Models/CertificateViewModel.cs

```
@@ -17,10 +29,35 @@ public GranularCertificate ToProto()
             }
         };
 
-        return new GranularCertificate
+        var res = new GranularCertificate
         {
             FederatedId = fedId,
-            Quantity = (uint)Quantity
+            Quantity = (uint)Slices.Sum(x => x.Quantity),
+            End = Timestamp.FromDateTimeOffset(EndDate),
+            Start = Timestamp.FromDateTimeOffset(StartDate),
+            GridArea = GridArea,
+            Type = ToDto(CertificateType),
         };
+
+        foreach (var atr in Attributes)
+        {
+            res.Attributes.Add(new V1.Attribute { Key = atr.Key, Value = atr.Value });
+        }
+
+        return res;
+    }
+
+    private V1.GranularCertificateType ToDto(GranularCertificateType type)
+    {
+        if (type == GranularCertificateType.Production)
+            return V1.GranularCertificateType.Production;
+
+        if (type == GranularCertificateType.Consumption)
+            return V1.GranularCertificateType.Consumption;
+
+        if (type == GranularCertificateType.Invalid)
+            return V1.GranularCertificateType.Invalid;
```

```suggestion
```

Remove and let the exception be thrown since it is invalid.